packages/PyTrilinos/src/Teuchos_Comm.i

```diff
@@ -610,40 +610,17 @@ else:
 MpiComm = MpiComm_int
 %}
 
-///////////////////////////////////////////
-// Teuchos.DefaultComm support under MPI //
-///////////////////////////////////////////
-%pythoncode
-%{
-
-class DefaultComm:
-    "Encapsulate the default global communicator"
-    __defaultComm = MpiComm(mpiCommunicator)
-    @classmethod
-    def getComm(cls):
-        "Return the default global communicator"
-        return cls.__defaultComm
-%}
-
-#else
-
-////////////////////////////////////////////////////////////////////////////////
-// The following code is implemented if HAVE_MPI is not defined
-////////////////////////////////////////////////////////////////////////////////
+#endif
 
-/////////////////////////////////////////////
-// Teuchos.DefaultComm support without MPI //
-/////////////////////////////////////////////
+/////////////////////////////////
+// Teuchos.DefaultComm support //
+/////////////////////////////////
+#define MPI_VERSION 2
```

Comment:
Are you sure you want to do this?  Trilinos only tests now with MPI implementations that have `MPI_VERSION` at least 3.

Reply:
This was from a while ago... I think it was a problem where SWIG needed it to be defined, and extracting just `MPI_VERSION` from `mpi.h` was too much of a hassle.  I can probably just change it to `3` if that is what Trilinos uses.

Reply:
@wfspotz It is definitely 3 is all of our tests.  Not sure if this matters much for PyTrilinos, though, so I'll just approve this PR for now.

Reply:
Nice catch.  This was from a while ago ... SWIG needs `MPI_VERSION` to be defined, but having it parse all of `mpi.h` to get it is not really practical.  It should just be changed to 3.

I discovered a number of cosmetic things in PyTrilinos that needed to be updated as well: the Developers Guide has changed because of the new infrastructure, URLs in docstrings needed to be changed, an unused script from the old infrastructure needed to be deleted, etc.  So my update will include `#define MPI_VERSION 3` as well as a bunch of other documentation-related things.